### DISCUSSION TITLE
Custom Role abilities - create repo

### DISCUSSION THREAD

**Discussion by quilter-danielbutcher** [community/community#51922] [New to GitHub]
I am a member of an organisation on github and I am trying to investigate the ability to create repositories when users don't have admin on the organisation.

We are hoping to create custom roles which we can apply to individual users so that they can create repositories without giving them organisational admin access.
Are you able to advise if this is possible?

**Comment by SMelidoni** [comment]
Hi there 👋🏽

Yes, it is possible to grant non-admin users the ability to create repositories within an organisation on GitHub without giving them full organisational admin access.

GitHub provides a feature called "Repository creation permissions" that allows you to specify which users or teams within your organisation can create repositories. You can grant this permission at the organisation level or at the team level.

To set this up:

1. Go to your organisation's settings on GitHub.
2. Click on "Member privileges" in the left-hand menu.
3. Under "Repository creation", select "Allow members to create repositories". 
4. Choose whether you want this permission to apply to all members of the organisation or only to specific teams.
5. If you chose to apply this permission to specific teams, click on "Teams" and select the teams you want to grant the permission to.
6. Click on "Save" to save your changes.

After you've set this up, the users or teams you've granted repository creation permission to will be able to create repositories within the organisation without requiring full organisational admin access.

Note that users with repository creation permissions can only create repositories within the organization's repositories limit. If the organisation has reached its limit, a user with repository creation permissions will not be able to create a new repository until an existing repository is deleted or the organisation's plan is upgraded.
